### PR TITLE
[cloud-aws] Move integration tests to unit tests

### PR DESCRIPTION
**Pull request by dadoonet**
We now use a `AwsEc2ServiceMock` which simulates responses from AWS API.

We can test with this that settings are really used as expected.

I just started to implement some tests. More to come!

I also removed Ec2DiscoveryITest as the goal of this class was only to check that when we start elasticsearch with this plugin, elasticsearch works fine.
We don't need that anymore as we now have RestIT which do that right (and better)!

Unsure if we want it in 2.0.0.Beta1 or later...

### CODE REVIEW

**Comment by dadoonet** [comment]
@s1monw wanna review this change?

**Comment by dadoonet** [comment]
@bleskes Could you help me on this?

File: plugins/cloud-aws/src/test/java/org/elasticsearch/discovery/ec2/AmazonEC2Mock.java [hunk 1]
```diff
@@ -0,0 +1,1358 @@
...
+                logger.debug("--> instance filtered");
+            }
+        }
+
+        return new DescribeInstancesResult().withReservations(
+                new Reservation().withInstances(filteredInstances)
+        );
+    }
+
+    // Not implemented methods in Mock
```
Comment (bleskes) [inline plugins/cloud-aws/src/test/java/org/elasticsearch/discovery/ec2/AmazonEC2Mock.java]: can we throw an UnsupportedOperationException in all those methods to make sure no once uses them?

File: plugins/cloud-aws/src/test/java/org/elasticsearch/discovery/ec2/Ec2DiscoveryTest.java [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+
+    @BeforeClass
+    public static void createThreadPool() {
+        threadPool = new ThreadPool(Ec2DiscoveryTest.class.getName());
+    }
+
+    @AfterClass
+    public static void stopThreadPool() {
+        if (threadPool !=null) {
+            threadPool.shutdownNow();
```
Comment (bleskes) [inline plugins/cloud-aws/src/test/java/org/elasticsearch/discovery/ec2/Ec2DiscoveryTest.java]: iuse org.elasticsearch.threadpool.ThreadPool#terminate(org.elasticsearch.threadpool.ThreadPool, long, java.util.concurrent.TimeUnit) ?

**Comment by bleskes** [comment]
I took a look and left some minor comments. I'm not expert in this code nor what it is expected to do, so I'm not sure how much it's worth but LGTM. I did expect to find some kind of integration test with ES's ZenDiscovery . Maybe use the fact we can mock the AWS services and try to form a cluster?

**Comment by dadoonet** [comment]
@bleskes I adressed your comments. Thanks!
I also rebased on latest 2.0 branch.

@rjernst Do you think you can also give a look to this?

**Comment by dadoonet** [comment]
Closed by fe74219 in 2.x branch.
Will follow up with another PR for master branch because of the split between s3 and ec2.

**Pull request closed without merging**